int32 mode
float32 time_delay
---
bool success
